Amazon.QLDB.Driver.Tests/AsyncQldbDriverTests.cs

```diff
@@ -0,0 +1,361 @@
+        private static readonly byte[] digest = { 89, 49, 253, 196, 209, 176, 42, 98, 35, 214, 6, 163, 93,
+            141, 170, 92, 75, 218, 111, 151, 173, 49, 57, 144, 227, 72, 215, 194, 186, 93, 85, 108,
+        };
+
+        [TestInitialize]
+        public void SetupTest()
+        {
+            mockClient = new Mock<AmazonQLDBSessionClient>();
+            var sendCommandResponse = new SendCommandResponse
+            {
+                StartSession = new StartSessionResult
+                {
+                    SessionToken = "testToken"
+                },
+                StartTransaction = new StartTransactionResult
+                {
+                    TransactionId = "testTransactionIdddddd"
+                },
+                ExecuteStatement = new ExecuteStatementResult
+                {
+                    FirstPage = new Page
+                    {
+                        NextPageToken = null,
+                        Values = new List<ValueHolder>()
+                    }
+                },
+                CommitTransaction = new CommitTransactionResult
+                {
+                    CommitDigest = new MemoryStream(digest)
+                },
+                ResponseMetadata = new ResponseMetadata
+                {
+                    RequestId = "testId"
+                }
+            };
+            mockClient.Setup(x => x.SendCommandAsync(It.IsAny<SendCommandRequest>(), It.IsAny<CancellationToken>()))
+                    .Returns(Task.FromResult(sendCommandResponse));
+            builder = AsyncQldbDriver.Builder()
+                .WithLedger("testLedger")
+                .WithRetryLogging()
+                .WithLogger(NullLogger.Instance)
+                .WithAWSCredentials(new Mock<AWSCredentials>().Object)
+                .WithQLDBSessionConfig(new AmazonQLDBSessionConfig());
+        }
+
+        [TestMethod]
+        public void TestAsyncBuilderGetsANotNullObject()
+        {
+            Assert.IsNotNull(builder);
+        }
+
+        [TestMethod]
+        public void TestAsyncWithPoolLimitArgumentBounds()
+        {
+            AsyncQldbDriver driver;
+        
+            // Default pool limit
+            driver = builder.Build();
+            Assert.IsNotNull(driver);
+        
+            // Negative pool limit
+            Assert.ThrowsException<ArgumentException>(() => builder.WithMaxConcurrentTransactions(-4));
+        
+            driver = builder.WithMaxConcurrentTransactions(0).Build();
+            Assert.IsNotNull(driver);
+            driver = builder.WithMaxConcurrentTransactions(4).Build();
+            Assert.IsNotNull(driver);
+        }
+        
+        [TestMethod]
+        public void TestAsyncQldbDriverConstructorReturnsValidObject()
+        {
+            var driver = new AsyncQldbDriver(
+                new AsyncSessionPool((ct) => Session.StartSessionAsync("ledgerName", mockClient.Object, NullLogger.Instance, ct),
```

Comment:
Are the parentheses around `(ct)` necessary? 
If it can be removed, there are other instances in the file.

Reply:
Removed.